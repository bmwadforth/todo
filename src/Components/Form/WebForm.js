import React from "react";
import PropTypes from "prop-types";

export default function WebForm(props) {
    return (
        <form onSubmit={props.onSubmit}>
            <div className="web-form">
                {props.children}
                <button type="submit">{props.actionTitle}</button>
            </div>
        </form>
    )
}

WebForm.propTypes = {
    onSubmit: PropTypes.func.isRequired,
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    actionTitle: PropTypes.string.isRequired
};
