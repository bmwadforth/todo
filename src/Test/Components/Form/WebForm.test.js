import React from 'react';
import {shallow, configure} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import WebForm from "../../../Components/Form/WebForm";
configure({adapter: new Adapter()});

describe("WebForm Component", () => {
    it("Should Match Snapshot", () => {
        const wrapper = shallow(<WebForm actionTitle="Test" onSubmit={() => {}}/>);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
