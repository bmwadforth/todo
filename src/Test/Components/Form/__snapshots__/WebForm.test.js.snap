// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebForm Component Should Match Snapshot 1`] = `"<form><div class=\\"web-form\\"><button class=\\"button button-primary\\" type=\\"submit\\">Test</button></div></form>"`;
